modules/apps/static/portal-file-install/portal-file-install-test/src/testIntegration/java/com/liferay/portal/file/install/deploy/test/FileInstallConfigTest.java

```diff
@@ -129,6 +135,59 @@ public void testConfigurationArrayValues() throws Exception {
 			(String[])properties.get("configUntypedStringArray"));
 	}
 
+	@Test
+	public void testConfigurationDeprecatedFileExtension() throws Exception {
+		String configurationPid = _CONFIGURATION_PID_PREFIX.concat(
+			".testDummy");
+		String configurationPidDeprecated = _CONFIGURATION_PID_PREFIX.concat(
+			".testConfigurationDeprecatedFileExtension");
+
+		String content = "testKey=\"testValue\"";
+		String contentDeprecated = "testKeyDeprecated=\"testValueDeprecated\"";
+
+		_configurationPath = Paths.get(
+			PropsValues.MODULE_FRAMEWORK_CONFIGS_DIR,
+			configurationPid.concat(".config"));
+
+		Path configPathDeprecated = Paths.get(
+			PropsValues.MODULE_FRAMEWORK_CONFIGS_DIR,
+			configurationPidDeprecated.concat(".cfg"));
+
+		try (CaptureAppender captureAppender =
+				Log4JLoggerTestUtil.configureLog4JLogger(
+					"com.liferay.portal.file.install.internal.configuration." +
+						"ConfigurationFileInstaller",
+					Level.WARN)) {
+
+			Files.write(
+				configPathDeprecated,
+				contentDeprecated.getBytes(Charset.defaultCharset()));
+
+			_configuration = _createConfiguration(configurationPid, content);
+
+			List<LoggingEvent> loggingEvents =
+				captureAppender.getLoggingEvents();
+
+			LoggingEvent loggingEvent = loggingEvents.get(0);
+
+			String actualWarningMessage = (String)loggingEvent.getMessage();
+
+			String expectedWarningMessage = StringBundler.concat(
+				"Unable to install .cfg file ", configPathDeprecated.toString(),
+				", please use .config file instead.");
+
+			Assert.assertEquals(expectedWarningMessage, actualWarningMessage);
+
+			Dictionary<String, Object> properties =
+				_configuration.getProperties();
+
+			Assert.assertNull(properties.get("testKeyDeprecated"));
```

Comment:
You need to check `testKeyDeprecated` on the cfg configuration, not the config one. The purpose of the check is to make sure that the cfg configuration actually didn't get loaded